code/class/registry/registry.php

```diff
@@ -129,7 +136,54 @@ public function alias($class, $alias)
     }
 
     /**
-     * Get a list of all the identifier aliases
+     * Get a specific locator from the class namespace
+     *
+     * @return string|false The name of the locator or FALSE if no locator could be found
+     */
+    public function getLocator($class)
+    {
+        $result = false;
+
+        if(!false == $pos = strrpos($class, '\\'))
+        {
+            $namespace = substr($class, 0, $pos);
+
+            if(isset($this->_namespaces[$namespace])) {
+                $result = $this->_namespaces[$namespace];
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * Set a specific locator based on a class namespace
+     *
+     * @return  ClassRegistry
+     */
+    public function setLocator($class, $locator)
+    {
+        if(!false == $pos = strrpos($class, '\\'))
```

Comment:
Sure, it will increase performance but it also introduces this new pattern that wasn't present before. Previously multiple locators could be registered for the same namespace, now that is not possible. If that's intended then roll on 👍

> On 11 Apr 2016, at 22:49, Johan Janssens <notifications@github.com> wrote:
> 
> In code/class/registry/registry.php:
> 
> > +            if(isset($this->_namespaces[$namespace])) {
> > +                $result = $this->_namespaces[$namespace];
> > +            }
> > +        }
> > +
> > +        return $result;
> > +    }
> > +
> > +    /**
> > +     * Set a specific locator based on a class namespace
> > +     *
> > +     * @return  ClassRegistry
> > +     */
> > +    public function setLocator($class, $locator)
> > +    {
> > +        if(!false == $pos = strrpos($class, '\\'))
> Oli, the scope of this ticket is to improve the class locator performance and prevent unneeded loops through the locator queue. The problem you point out is not in this scope.
> 
> With that said, you can do this, using class_alias or ClassLoader::registerAlias(). If you want to use a dynamic approach you create a specialised locator, prepend it and make it fallback for classes it cannot handle, either by extending or composing another locator.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>